src/main/resources/avro/genotypephenotype.avdl

```diff
@@ -0,0 +1,204 @@
+  Anatomy (Uberon): http://www.ontobee.org/browser/index.php?o=uberon
+  */
+  OntologyTerm environmentType;
+
+  /**
+	A textual description of the environment. This is used to complement
+	the structured description in the environmentType field
+  */
+  union { null, string } description = null;
+
+}
+
+
+/**
+An association to a phenotype and related information.
+This record is intended primarily to be used in conjunction with variants, but
+the record can also be composed with other kinds of entities such as diseases
+*/
+record PhenotypeInstance {
+  /** The Phenotype ID. */
+  union { null, string } id;
+
+  /** HPO is recommended */
+  OntologyTerm type;
+
+  /**
+  PATO is recommended.  Often this qualifier might be for abnormal/normal,
+  or severity.
+  For example, severe: http://purl.obolibrary.org/obo/PATO_0000396
+  or abnormal: http://purl.obolibrary.org/obo/PATO_0000460
+  */
+  union { null, array<OntologyTerm> } qualifier = null;
+
+  //TODO: also allow quantitative recording?
+  /**
+  HPO is recommended, for example, subclasses of
+  http://purl.obolibrary.org/obo/HP_0011007
+  */
+  union { null, OntologyTerm } ageOfOnset = null;
+
+  /**
+  A textual description of the phenotype. This is used to complement the
+  structured phenotype description in the type field.
+  */
+  union { null, string } description = null;
+
+}
+
+
+/**
+Evidence for the phenotype association.
+This is also a stub for further expansion.  We should consider moving this into
+it's own schema.
+*/
+record Evidence {
+
+  /** ECO or OBI is recommended */
+  OntologyTerm evidenceType;
+
+  /**
+	A textual description of the evidence. This is used to complement the
+	structured description in the evidenceType field
+	*/
+  union { null, string } description = null;
+
+  /** Additional annotation data in key-value pairs. */
+  map<array<string>> info = {};  
+}
+
+/**
+An association between one or more genomic features and a phenotype.
+The instance of association allows us to link a feature to a phenotype,
+multiple times, each bearing potentially different levels of confidence,
+such as resulting from alternative experiments and analysis.
+*/
+record FeaturePhenotypeAssociation {
+  string id;
+
+  /**
+  The ID of the PhenotypeAssociationSet this FeaturePhenotypeAssociation
+  belongs to.
+  */
+  string phenotypeAssociationSetId;
+
+  /**
+    The set of features of the organism that bears the phenotype.
+    This could be as complete as a full complement of variants,
+    or as minimal as the confirmed variants that are known causation
+    for the annotated phenotype.
+    Examples of features could be variations at the nucleotide level,
+    large rearrangements at the chromosome level, or relevant epigenetic
+    markers.  Relevant genomic feature types are suggested to be
+    those typed in the Sequence Ontology (SO).
+
+    The feature set can have only one item, and must not be null.
+  */
+  array<Feature> features;
```

Comment:
GA4GH Variant records speak about specific mutations that occur in specific samples. A phenotypic association is usually connected to the gene, which is a 'concept' shared across a multiple samples. So you have a variant in a gene that then has an association to a phenotype.